README.md

```diff
@@ -512,6 +518,27 @@ The `React\Promise\PromisorInterface` provides a common interface for objects
 that provide a promise. `React\Promise\Deferred` implements it, but since it
 is part of the public API anyone can implement it.
 
+### AsyncInterop\Promise compatibility
+
+The [`PromiseInterface`](#promiseinterface) extends the
+[`AsyncInterop\Promise`](https://github.com/async-interop/promise) interface and
+thereby all promise implementations from React/Promise implement its `when()`
+method.
+
+The `AsyncInterop\Promise` interface is intended for interoperability with other
+libraries and their combinator and conversion functions.
+
+Note, that although the interface is named *Promise*, it has nothing to do with
+promises commonly known from the
```

Comment:
> *"Why the hell is it called promise if it's something entirely different?!"*

I think this is a very important point here: To me, this whole PR seems to convey the notion that *this* library implements a concept commonly known as "promises", while async-interop/promise *only* offer a low-level API that aims to ease interoperability between different promise vendors.

> We had a really long discussion about the naming in async-interop/promise#4 and another one to switch back to Promise, but I can't find that one right now.

This isn't *exactly transparent* to me either, so I'd very much appreciate if you happen to find this and could link to this here :+1: 

> […] we kept calling it Promise while talking about it, because it's just that, just with a different consumer API.

I understand the reasoning – but would disagree from a consumer perspective. If there's project A (e.g. this project) that describes "promises" and the interop standard that describes a different API then it's not the same from a consumer perspective.

This means that project A should either ditch its description and adopt the interop API or the other way around (which I don't see happening any time soon).